router/userRouter.js

```diff
@@ -1,7 +1,12 @@
 const router = require("express").Router();
-const { validateFields } = require("../middlewares/userMiddleWare")
+const { validateFields, validatePostFields, validateFieldErrors } = require("../middlewares/userMiddleWare")
 const userController = require("../controller/userController")
+const postController = require("../controller/postController")
+const { uploadImageToFirebase } = require("../controller/uploadImage");
+const { header } = require("express-validator");
 router.post("/register", validateFields, userController.userRegister)
 router.post("/create-user-firebase-token", userController.createFirebaseToken)
 router.post("/update-user", userController.updateUserDetails)
+router.post("/create-new-post", [validatePostFields, validateFieldErrors], postController.createPost)
+router.post("/upload-image", [[header("uid").notEmpty()], validateFieldErrors, uploadImageToFirebase.single("image")], userController.uploadImage)
```

Comment:
your function does not belongs to a user, its for any image upload create another controller if it, it should not be un user controller @ashok-choudhary-0